File: scratch_query.py
import urllib.request
import json

url = "https://avvitujfdhjqzcuhfpex.supabase.co/rest/v1/bacaan?select=*"
headers = {
    "apikey": "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJzdXBhYmFzZSIsInJlZiI6ImF2dml0dWpmZGhqcXpjdWhmcGV4Iiwicm9sZSI6ImFub24iLCJpYXQiOjE3NjkxNTAxMzAsImV4cCI6MjA4NDcyNjEzMH0.FZfYw1Z8yNskYp5uRSlnXaMcRHH5ATlFXWitUwHU7wI",
    "Authorization": "Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJzdXBhYmFzZSIsInJlZiI6ImF2dml0dWpmZGhqcXpjdWhmcGV4Iiwicm9sZSI6ImFub24iLCJpYXQiOjE3NjkxNTAxMzAsImV4cCI6MjA4NDcyNjEzMH0.FZfYw1Z8yNskYp5uRSlnXaMcRHH5ATlFXWitUwHU7wI"
}

req = urllib.request.Request(url, headers=headers)
try:
    with urllib.request.urlopen(req) as response:
        res = response.read().decode("utf-8")
        data = json.loads(res)
        print("Query success! Found", len(data), "records.")
        for item in data:
            if 'Nas' in item['content'] or 'Nas' in item['title'] or 'نسألك' in item['content'] or 'نَسْئَلُكَ' in item['content']:
                print('--- TITLE:', item['title'])
                print('--- CONTENT:')
                print(item['content'])
                print('-------------------')
except Exception as e:
    print("Error querying database:")
    if hasattr(e, "read"):
        print(e.read().decode("utf-8"))
    else:
        print(str(e))

File: scratch_query.js
import { createClient } from "@supabase/supabase-js";

const supabaseUrl = 'https://avvitujfdhjqzcuhfpex.supabase.co';
const supabaseKey = 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJzdXBhYmFzZSIsInJlZiI6ImF2dml0dWpmZGhqcXpjdWhmcGV4Iiwicm9sZSI6ImFub24iLCJpYXQiOjE3NjkxNTAxMzAsImV4cCI6MjA4NDcyNjEzMH0.FZfYw1Z8yNskYp5uRSlnXaMcRHH5ATlFXWitUwHU7wI';

const supabase = createClient(supabaseUrl, supabaseKey);

async function run() {
    console.log("Checking allowed_alumni columns...");
    const { data, error } = await supabase.from("allowed_alumni").select("*").limit(1);
    if (error) {
        console.error("Error fetching allowed_alumni:", error);
    } else if (data && data.length > 0) {
        console.log("Columns:", Object.keys(data[0]));
        console.log("Sample Data:", data[0]);
    } else {
        console.log("No records found in allowed_alumni.");
    }
}

run();
